src/core/lib/iomgr/tcp_posix.cc

```diff
@@ -836,11 +844,21 @@ static void maybe_make_read_slices(grpc_tcp* tcp) {
     int target_length = static_cast<int>(tcp->target_length);
     int extra_wanted =
         target_length - static_cast<int>(tcp->incoming_buffer->length);
+    if (tcp->last_read_completed) {
+      // Set it to false again to start the next block of reads
+      tcp->last_read_completed = false;
+      // Reset last_min_read_chunk_size for the next block of reads
+      tcp->last_min_read_chunk_size = tcp->min_read_chunk_size;
```

Comment:
nit: maybe rename `curr_min_read_chunk_size` and `starting_min_read_chunk_size` or something similar?

Reply:
Changed it now to curr_min_read_chunk_size and curr_read_completed